src/Parser/Rogue/Subtlety/Modules/RogueCore/ComboPointTracker.js

```diff
@@ -1,16 +1,32 @@
+import SPELLS from 'common/SPELLS';
 import Combatants from 'Parser/Core/Modules/Combatants';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
 
 import ResourceTracker from '../ResourceTracker/ResourceTracker';
 
 class ComboPointTracker extends ResourceTracker {
   static dependencies = {
     combatants: Combatants,
+		spellUsable: SpellUsable,
   };
 
   on_initialized() {
     this.resourceType = 4;
     this.resourceName = "Combo Points";
   }
+
+  onSpent(spent)  {    
+    const combatant = this.combatants.selected;
+
+    if (this.spellUsable.isOnCooldown(SPELLS.SHADOW_DANCE.id)) {
+      const reduction = combatant.hasTalent(SPELLS.ENVELOPING_SHADOWS_TALENT.id) ? 2500 : 1500;
+
```

Comment:
Move this ``const reduction`` into your ``on_initialized`` please. Alternatively you can create a constant called REDUCTION above the class indicator, and then in ``on_initialized`` add 1000 to it if the player has the talent, that would be the most consistent with other similiar modules.